src/main/java/com/atlassian/db/replica/internal/SQLOpeartionType.java

```diff
@@ -0,0 +1,50 @@
+package com.atlassian.db.replica.internal;
+
+public class SQLOpeartionType {
+
+    public static enum SQLType {
+        WRITE, READ_ONLY, NOT_KNOWN
+    }
+
+    private final SqlFunction sqlFunction;
+
+    public SQLOpeartionType(SqlFunction sqlFunction) {
+        this.sqlFunction = sqlFunction;
+    }
+
+    /**
+     * Arbitrially judge based on sql param whether sql is WRITE
+     */
+    public boolean whetherSQLisWrite(String sql){
```

Comment:
```suggestion
    public boolean writesToDatabase(String sql){
```

Reply:
No! writesToDatabase might happen for READONLY Endpoint when Replica inconsistent, db-replica has some bug / is broken. But this doesn't change the type of SQL query READ/WRITE

Reply:
I'm confused. I'm just suggesting cleaner English without changing semantics.
Call site:
```java
var type = ...
if (type.writesToDatabase("UPDATE ...")) { ...
```

It's still procedural style, so OOP style might be clearer:
```java
var query = new SqlQuery("UPDATE ...")
if (query.writesToDatabase()) { ...
```

Reply:
Ok. We misunderstood.  I've had different expectation from this.  You stick to the implementation, I'm still not sure about proper meaning of `isWrite`

Reply:
There are many comments. Some affect big portions of the diff. Those are under discussion for longer.
Some are small, like typos, better naming, local refactoring. Those are quickly agreed upon and can be fixed immediately.
Even if a wide change will affect the small changes, the small ones are are worth doing anyway (just to reduce the comments).